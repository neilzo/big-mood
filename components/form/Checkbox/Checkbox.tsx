import React, { useState } from 'react';
import { View, TouchableWithoutFeedback, StyleSheet } from 'react-native';

import Icon from '../../Icon/Icon';
import colorVariables from '../../colorVariables';

interface Props {
  containerStyle?: object;
  onChange: (selected: boolean) => void;
  checked?: boolean;
}
const Checkbox = ({
  containerStyle,
  onChange = () => {},
  checked = false,
}: Props) => {
  const [selected, setSelected] = useState(checked);
  return (
    <TouchableWithoutFeedback
      onPress={() => {
        const newSelected = !selected;
        setSelected(newSelected);
        onChange(newSelected);
      }}
    >
      <View style={containerStyle}>
        {!selected && <Icon name="checkbox-blank-outline" size={20} />}
        {selected && (
          <Icon
            name="checkbox-marked-outline"
            size={20}
            color={colorVariables.selected}
          />
        )}
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({});

export default Checkbox;
