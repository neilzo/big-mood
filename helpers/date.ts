import moment from 'moment';

const getPrettyDate = (date: Date) => moment(date).format('MMM Do YYYY');

const getPrettyTime = (date: Date) => moment(date).format('h:mma');

export default { getPrettyDate, getPrettyTime };
