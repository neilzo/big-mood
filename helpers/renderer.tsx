import 'react-native';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

export const createComponent = (component: any) => {
  return renderer.create(component);
};

// export const createComponentWithRedux = (children: any, props = {}) => {
//   return renderer.create(<Provider {...props}>{children}</Provider>);
// };
