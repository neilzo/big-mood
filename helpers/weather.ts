interface WeatherResults {
  weather: Array<{ main: string; icon: string }>;
  main: {
    temp: number;
  };
}

const kelvinToFarenheit = (kelvin: number) => {
  return Math.round(kelvin * (9 / 5) - 459.76);
};

const computeOverallWeather = (weatherResults: WeatherResults) => ({
  icon: weatherResults.weather[0].icon,
  description: weatherResults.weather[0].main,
  temperature: kelvinToFarenheit(weatherResults.main.temp),
});

export { kelvinToFarenheit, computeOverallWeather };
