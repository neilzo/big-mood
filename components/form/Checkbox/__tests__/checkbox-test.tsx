import React from 'react';
import Checkbox from '../Checkbox';
import { createComponent } from '../../../../helpers/renderer'; // todo resolve imports to avid that ugly thang

jest.mock('../../../Icon/Icon', () => 'Icon');

const mockProps = {
  onChange: () => {}
};

describe('Checkbox', () => {
  it('renders correctly', () => {
    expect(
      createComponent(<Checkbox {...mockProps} />).toJSON()
    ).toMatchSnapshot();
  });

  it('renders with checked', () => {
    expect(
      createComponent(<Checkbox {...mockProps} checked />).toJSON()
    ).toMatchSnapshot();
  });
});
