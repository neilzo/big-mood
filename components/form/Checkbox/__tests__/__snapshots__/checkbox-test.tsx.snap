// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Icon
    name="checkbox-blank-outline"
    size={20}
  />
</View>
`;

exports[`Checkbox renders with checked 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Icon
    color="#007aff"
    name="checkbox-marked-outline"
    size={20}
  />
</View>
`;
